main/src/ca/uwaterloo/flix/util/ParOps.scala

```diff
@@ -17,30 +17,41 @@
 package ca.uwaterloo.flix.util
 
 import java.util
-import java.util.concurrent.{Callable, Executors}
-
+import java.util.concurrent.{Callable, CountDownLatch}
 import scala.jdk.CollectionConverters._
 import scala.collection.parallel._
 import ca.uwaterloo.flix.api.Flix
 
+import scala.reflect.ClassTag
+
 object ParOps {
 
   /**
     * Apply the given function `f` to each element in the list `xs` in parallel.
     */
   @inline
-  def parMap[A, B](xs: Iterable[A])(f: A => B)(implicit flix: Flix): Iterable[B] = {
-    // Build the parallel array.
-    val parArray = xs.toParArray
-
-    // Configure the task support.
-    parArray.tasksupport = flix.forkJoinTaskSupport
-
-    // Apply the function `f` in parallel.
-    val result = parArray.map(f)
+  def parMap[A, B: ClassTag](xs: Iterable[A])(f: A => B)(implicit flix: Flix): Iterable[B] = {
+    // Compute the size of the input and construct a new empty array to hold the result.
+    val size = xs.size
+    val out: Array[B] = new Array(size)
+
+    // Construct a new count down latch to track the number of threads.
+    val latch = new CountDownLatch(size)
+
+    // Iterate through the elements of `xs`. Use a local variable to track the index.
+    var idx = 0
```

Comment:
zipWithIndex could avoid some mutation

Reply:
I actually switched from zipWithIndex __because it improved performance!!__